android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java

```diff
@@ -80,6 +80,8 @@ public void onNetworkItemSelected(NetworkSelectorItem networkSelectorItem) {
                         CoinType.ETH, (success) -> {
                             if (!success) {
                                 Log.e(TAG, "Could not set network");
+                            } else {
+                                finish();
```

Comment:
If the activity is not finished when there is a failure then nothing will happen but the selection on the UI is updated which means neither it will auto close nor give information of error to user while the nectwork is changed on screen.

I think would be better to revert the selected network change in case of failure and probably show a toast to the user about a failure.

cc: @SergeyZhukovsky @jamesmudgett any thoughts?

Reply:
This should be addressed by the issue [22535](https://github.com/brave/brave-browser/issues/22535)